shell/src/main/java/alluxio/shell/command/CpCommand.java

```diff
@@ -551,10 +551,19 @@ private void copyToLocal(AlluxioURI srcPath, AlluxioURI dstPath) throws AlluxioE
    */
   private void copyFileToLocal(AlluxioURI srcPath, AlluxioURI dstPath)
       throws AlluxioException, IOException {
-    File dstFile = new File(dstPath.getPath());
+    File dstFile = new File(dstPath.getAbsolutePath());
     String randomSuffix =
         String.format(".%s_copyToLocal_", RandomStringUtils.randomAlphanumeric(8));
-    File tmpDst = new File(dstFile.getAbsolutePath() + randomSuffix);
+    File tmpDst;
+    File outputFile;
+    if (dstFile.isDirectory()) {
+      tmpDst = new File(PathUtils.concatPath(dstFile.getAbsolutePath(),
```

Comment:
we can simplify this a little by using if-else to determine `outputFile`, then set `tmpDst` to be `outputFile` plus a random suffix outside of the if-else